File: src/test/java/com/hpe/ossm/jcluster/test/CMService.java
package com.hpe.ossm.jcluster.test;

import akka.actor.*;
import com.hpe.ossm.jcluster.ClusterNode;
import com.hpe.ossm.jcluster.ServiceEntryActor;
import java.util.HashMap;

public class CMService extends ServiceEntryActor {

    @Override
    public void preStart() throws  Exception {
        //override the parameters
        serviceName = "CM";
        setDependServices(null);
        super.preStart();
    }

    @Override
    public void postStop()throws  Exception {
        super.postStop();
    }

    @Override
    public AbstractActor.Receive createReceive() {
        return super.createReceive()
                .orElse(
                        receiveBuilder()
                                .match(String.class, m-> getSender().tell(cluster.selfAddress()+"|"+m, getSelf()))
                                .matchAny( o->System.out.println("Ignored message "+ o.toString()))
                                .build()
                );
    }
}

class  CMServiceApp extends ClusterNode {
    public CMServiceApp(String role){
        this.role=role;
    }
    public static void main(String[] args){
        CMServiceApp t=new CMServiceApp("CMService");
        ActorSystem sys=t.getActorSystem(null);
        sys.actorOf(Props.create(CMService.class),"CM");
    }
}

File: src/test/scala/com/hpe/ossm/scluster/test/CMService.scala
package com.hpe.ossm.scluster.test

import akka.actor.{DeadLetter, Props}
import akka.cluster.Cluster
import com.hpe.ossm.scluster.management.DeadLetterListener
import com.hpe.ossm.scluster.{ClusterNode, ServiceEntryActor}

class CMService extends ServiceEntryActor("CM", null) {
    val cluster=Cluster(context.system)
    override def preStart(): Unit = {
        println(s"CMService starting")
        super.preStart()
    }

    override def postStop(): Unit = super.postStop()


    override def receive: Receive = {
        super.receive.orElse {
            case s: String =>
                println(s"[CM]received "+s)
                sender ! cluster.selfAddress + "|" + s
            case s: Any => println(s"Ignored message ${s.toString}")
        }
    }
}

class CMServiceApp extends ClusterNode("CMService", null)

object CMServiceNode {
    def main(args: Array[String]): Unit = {
        val t = new CMServiceApp
        val ref = t.system.actorOf(Props(classOf[DeadLetterListener]), "deadLetterListener")
        t.system.eventStream.subscribe(ref, classOf[DeadLetter])
        t.system.actorOf(Props(classOf[CMService]), "CM")
    }
}
